rpmlint/checks/FileDigestCheck.py

```diff
@@ -34,46 +38,90 @@ def _get_digest_group_from_path(self, path):
                     pass
         return None
 
-    def check_binary(self, pkg):
+    def _in_digest_configuration_group(self, pkgfile):
+        if stat.S_ISDIR(pkgfile.mode):
+            return False
+        return self._get_digest_location_from_path(pkgfile.name)
+
+    def _check_symlinks(self, pkg):
         """
-        TODO: add comment
+        Check that all symlinks point to a correct location and that
+        symlinks are allowed in a configuration group.
         """
-        known_digests = self.package_digests.get(pkg.name)
+        result = True
         for filename, pkgfile in pkg.files.items():
-            group = self._get_digest_group_from_path(filename)
-            if not group:
+            group = self._get_digest_location_from_path(filename)
+            if not self._in_digest_configuration_group(pkgfile):
                 continue
-            if stat.S_ISDIR(pkgfile.mode):
+
+            if filename in pkg.ghost_files:
+                self.output.add_info('E', pkg, f'{group}-file-digest-ghost', filename)
+                result = False
                 continue
 
             if stat.S_ISLNK(pkgfile.mode) and not self.follow_symlinks_in_group[group]:
-                self.output.add_info('W', pkg, f'{group}-file-symlink', filename)
+                self.output.add_info('E', pkg, f'{group}-file-symlink', filename)
+                result = False
                 continue
 
             pkgfile = pkg.readlink(pkgfile)
             if not pkgfile:
-                self.output.add_info('W', pkg, f'{group}-file-symlink', filename, 'broken symlink')
-                continue
+                self.output.add_info('E', pkg, f'{group}-file-symlink', filename, 'broken symlink')
```

Comment:
The symlink is not necessarily broken, but probably a symlink outside the package. It might become necessary to support such situations. With rpmlint there is probably no way to check such cross-package symlinks, bet we would at least need to be able to whitelist such a case using the 'skip' algorithm.

Reply:
All right, I fixed that.